service/service_test.go

```diff
@@ -18,10 +18,44 @@ import (
 	"context"
 	"testing"
 
+	"github.com/golang/mock/gomock"
 	"github.com/google/go-cmp/cmp"
 	pb "github.com/kubeflow/metadata/api"
+	mlmd "github.com/kubeflow/metadata/ml_metadata"
+	"github.com/kubeflow/metadata/mocks"
 )
 
+const schemaDir = "../schema/alpha"
+
+func TestNewService(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockMLMDClient := mocks.NewMockMetadataStoreServiceClient(ctrl)
+	var mlmdTypeID int64 = 1
+	mockMLMDClient.EXPECT().PutArtifactType(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(context.Context, *mlmd.PutArtifactTypeRequest) (*mlmd.PutArtifactTypeResponse, error) {
+			id := mlmdTypeID
+			mlmdTypeID++
+			return &mlmd.PutArtifactTypeResponse{
+				TypeId: &id,
+			}, nil
+		}).AnyTimes()
+	s, err := NewService(mockMLMDClient, schemaDir)
+	if err != nil {
+		t.Fatalf("Failed to create a new service: %v", err)
+	}
+	mustInclude := []string{
+		"namespaces/kubeflow.org/kinds/data_set/versions/alpha",
```

Comment:
I think we agree it will be useful to have several more specific types for data set. but it is still good to have a general type to capture any data if no suitable specific one can be used.